### PR TITLE
22950 update dissolution jobs use new db versioning

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#22950

*Description of changes:*

- Replaced the code to initialize and use the new db versioning
- Updated configuration to support feature flags
- Added flags.json file
- Updated requirements.txt with necessary dependencies
- Switch the versioning Proxy

*Test:*
1. The Flag is on:
- Proxy Check
Screenshot 2024-12-04 at 1 11 06 AM
- Unit Test
Screenshot 2024-12-04 at 1 09 10 AM

2. The Flag is off:
- Proxy Check
Screenshot 2024-12-04 at 1 10 20 AM
 - Unit Test
Screenshot 2024-12-04 at 1 09 28 AM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: jobs/involuntary-dissolutions/config.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ def get_named_config(config_name: str = 'production'):
 class _Config(object):  # pylint: disable=too-few-public-methods
     """Base class configuration."""
 
+    SERVICE_NAME = 'dissolutions-job'
```
Comment (kzdev420) [inline jobs/involuntary-dissolutions/config.py:48]: Add service-name that can identify versioning flag

File: jobs/involuntary-dissolutions/flags.json [hunk 2]
```diff
@@ -2,6 +2,19 @@
...
+            "filer": false,
+            "entity-bn": true,
+            "digital-credentials": false,
+            "dissolutions-job": true,
+            "furnishings-job": true,
+            "emailer-reminder-job": false,
+            "update-colin-filings-job": false,
+            "future-effective-job": false,
+            "update-legal-filings-job": false
+        }
```
Comment (kzdev420) [inline jobs/involuntary-dissolutions/flags.json:18]: Add db-versioning flag values

File: jobs/involuntary-dissolutions/requirements.txt [hunk 3]
```diff
@@ -30,6 +30,9 @@ six==1.15.0
 urllib3==1.26.11
 requests==2.25.1
 cachelib==0.1.1
-git+https://github.com/bcgov/lear.git#egg=legal_api&subdirectory=legal-api
+# install updated legal-api from feature branch for development, 
+# TODO - change it back when the db versioning feature branch merged to main
+git+https://github.com/bcgov/lear.git@feature-db-versioning#egg=legal_api&subdirectory=legal-api
+# git+https://github.com/bcgov/lear.git#egg=legal_api&subdirectory=legal-api
```
Comment (kzdev420) [inline jobs/involuntary-dissolutions/requirements.txt:36]: Now, we'll pull legal-api's update from the feature-db-versioning branch, and switch back to the main later

File: jobs/involuntary-dissolutions/flags.json [hunk 4]
```diff
@@ -2,6 +2,19 @@
...
+            "emailer": true,
+            "filer": false,
+            "entity-bn": true,
+            "digital-credentials": false,
+            "dissolutions-job": true,
+            "furnishings-job": true,
+            "emailer-reminder-job": false,
+            "update-colin-filings-job": false,
+            "future-effective-job": false,
+            "update-legal-filings-job": false
```
Comment (eason-pan-bc) [inline jobs/involuntary-dissolutions/flags.json:17]: `future-effective-job` and `update-legal-filings-job` have been removed from FFs, you can check the current db-versioning FFs in legal-api (https://github.com/bcgov/lear/blob/feature-db-versioning/legal-api/flags.json)

**Review by eason-pan-bc — COMMENTED** [review]
Please consider updating the FFs value in the updated services:
- Queue Service:
    - [ ] Entity-BN
    - [ ] Entity-Emailer
- Jobs:
    - [ ] Furnishings Job

**Review by leodube-aot — APPROVED** [review]

**Review by AimeeGao — COMMENTED** [review]
Thanks for the detailed screenshots on the PR. 

I noticed from your screenshots that the dissolution job's stage 1 was skipped. 
Could you please provide additional test screenshots showing:
The state of DB versioning when stage 1 runs successfully.
The state of DB versioning when stage 2 runs, and etc.
Any DB operations that occur between these stages

This will help us understand the job's behavior and verify the versioning logic is working as expected.
Thx

File: jobs/involuntary-dissolutions/flags.json (hunk 4 above)
Comment (AimeeGao) [reply jobs/involuntary-dissolutions/flags.json]: Also, please consider removing the `update-colin-filings-job` flag here

**Review by eason-pan-bc — APPROVED** [review]
Code looks good to me.
Like Aimee mentioned, it would be great if there is actually a business from your local db can be picked up by the job to have a better verification.

**Review by AimeeGao — APPROVED** [review]
Thanks, LGTM(^-^)

**Pull request merged**